cmd/jaeger/internal/integration/grpc_test.go

```diff
@@ -7,43 +7,35 @@ import (
 	"testing"
 
 	"github.com/jaegertracing/jaeger/internal/storage/integration"
-	"github.com/jaegertracing/jaeger/ports"
 )
 
-type GRPCStorageIntegration struct {
-	E2EStorageIntegration
-
-	remoteStorage        *integration.RemoteMemoryStorage
-	archiveRemoteStorage *integration.RemoteMemoryStorage
-}
-
-func (s *GRPCStorageIntegration) initializeRemoteStorages(t *testing.T) {
-	s.remoteStorage = integration.StartNewRemoteMemoryStorage(t, ports.RemoteStorageGRPC)
-	s.archiveRemoteStorage = integration.StartNewRemoteMemoryStorage(t, ports.RemoteStorageGRPC+1)
-}
-
-func (s *GRPCStorageIntegration) closeRemoteStorages(t *testing.T) {
-	s.remoteStorage.Close(t)
-	s.archiveRemoteStorage.Close(t)
-}
-
-func (s *GRPCStorageIntegration) cleanUp(t *testing.T) {
-	s.closeRemoteStorages(t)
-	s.initializeRemoteStorages(t)
-}
-
 func TestGRPCStorage(t *testing.T) {
 	integration.SkipUnlessEnv(t, "grpc")
 
-	s := &GRPCStorageIntegration{
-		E2EStorageIntegration: E2EStorageIntegration{
-			ConfigFile:         "../../config-remote-storage.yaml",
-			SkipStorageCleaner: true,
+	remoteBackend := &E2EStorageIntegration{
+		ConfigFile:      "../../config-remote-storage-backend.yaml",
+		HealthCheckPort: 12133,
+		MetricsPort:     8887,
+		StorageIntegration: integration.StorageIntegration{
+			CleanUp: purge,
```

Comment:
`remoteBackend` is a helper component, the tests are being executed against `collector`. So cleanup will only be called on `collector`. However, if clean-up is meant to invoke the storage purger, it can only exist in the `remoteBackend` because that's where the memory storage resides. Since the cleanup (or rather `purge()`) works by making an RPC to the extension on a known port, it should work fine, you just need to have cleanup defined on the `collector` below, not here.

Reply:
@yurishkuro Thanks for the great explanation! Fixed.